### ISSUE TITLE
Auth.php threw uncaught DB error at refresh install

### ISSUE THREAD

**Issue by vauntmag** [delight-im/PHP-Auth#92]
Hi,

Got this great library working locally on MAMP and now experiencing very strange errors on hosting server...fresh install, var_dump shows $db and $auth working...but the moment I try to register a user, the following Fatal error was thrown...this did not happen on my local dev server before...

Any ideas? Any help would be greatly appreciated!

Fatal error: Uncaught exception 'Delight\Auth\DatabaseError' in /home/vhosts/{hostname}/{hostname}/vendor/delight-im/auth/src/Auth.php:1695 Stack trace: #0 /home/vhosts/{hostname}/{hostname}/vendor/delight-im/auth/src/Auth.php(230): Delight\Auth\Auth->throttle(Array, 1, 3600, 75) #1 /home/vhosts/{hostname}/{hostname}/test3.php(17): Delight\Auth\Auth->register('test@test.com', 'test123', Object(Closure)) #2 {main} thrown in /home/vhosts/{hostname}/{hostname}/vendor/delight-im/auth/src/Auth.php on line 1695

**Comment by ocram** [comment]
What database software are you using on the server? Is it the same that you also used locally?

Please make sure that your database schema is set up correctly and is up to date as well. The problem here occurs when the library tries to access the `users_throttling` table.

Finally, if you need to debug this, see https://github.com/delight-im/PHP-Auth/issues/91. For you, the line in question is `Auth.php:1695`. You could add the `$e` argument there.

**Comment by vauntmag** [comment]
Thank you for such quick reply!

Indeed, the problem is that the hosting service uses MariaDB and my local MAMP install uses MySQL.

Since this error happened in users-throttling, I tried next to disable throttle (just to see what happens) and a different error happened at user registration. So that basically shows me it is a MariaDB issue…I have grown quite attached to this library over the last 24 hours…fantastic clean code!! So instead of wrestling with MariaDB, I am now trying to install on a different hosting provider with MySQL and see what happens (of course it should work!).

Thank you for replying so quickly! …Have you ever seen these types of issues with MariaDB?

**Comment by vauntmag** [comment]
Ok...I  must have offended the SQL gods somehow...I did a fresh install on a server with the same MySQL version and PHP version as my local dev environment...a similar error has thrown...this time Delight\Auth\Administration->createUser('test@test.com', 'test123', NULL) #2 {main} thrown in /home/stamppro/public_html/vendor/delight-im/auth/src/UserManager.php on line 175

Any help would be deeply appreciated! I am at my wits end at the moment...

**Comment by vauntmag** [comment]
Ok...did exactly what you proposed with the extended error messages. Turns out that it was the hosting servers' problem :):) of course.  Thank you so much for all your help!

**Comment by ocram** [comment]
Thank you very much for your appreciation and for the further details on this issue!

It seems we should probably enable those extended error messages by default. This problem having come up just a few days ago and then now again is certainly no coincidence. This is now tracked in https://github.com/delight-im/PHP-Auth/issues/93.

Does this mean the library is now working for you with both MySQL and MariaDB? Or haven’t you tried again with MariaDB?

Can you explain what was wrong with the server? Was it a misconfiguration that could affect others as well?

**Comment by vauntmag** [comment]
Thank you so much for your help. The extra error messages really helped to nail down the problem! I

So long story short, I had to try a third host to get this to work. The first one was with MariaDB, I never did get to test it again. The second host had a connection problem: db can be reached, connection string was valid, but they had a problem with verification of the user credentials to access the db. They never did fix it for me. I went to a third host with MySQL5.6 and Php7.2 and all is well, worked like a charm just like it did on my local dev MAMP.

I think this is a freak situation….but the extra error messages really helped to convince hosting support that the issue is on their side, not with the code. So thank you very much again!
